landoapi/workers/landing_worker.py

```diff
@@ -303,35 +291,16 @@ def run_job(
                 self.notify_user_of_landing_failure(job)
                 return True
 
-            # Download all patches locally from S3.
-            patch_bufs = []
-            for revision_id, diff_id in job.landing_path:
-                try:
-                    patch_buf = patches.download(
-                        revision_id,
-                        diff_id,
-                        patch_bucket,
-                        aws_access_key=self.config["AWS_ACCESS_KEY"],
-                        aws_secret_key=self.config["AWS_SECRET_KEY"],
-                        endpoint_url=self.config["S3_ENDPOINT_URL"],
-                    )
-                except Exception as e:
-                    message = (
-                        f"Aborting, could not fetch {revision_id}, {diff_id} from S3."
-                    )
-                    logger.exception(message)
-                    job.transition_status(
-                        LandingJobAction.FAIL,
-                        message=message + f"\n{e}",
-                        commit=True,
-                        db=db,
-                    )
-                    self.notify_user_of_landing_failure(job)
-                    return True
-                patch_bufs.append((revision_id, patch_buf))
+            # Fetch all patches.
+            all_patches = []
+
+            for revision_id, _diff_id in job.landing_path:
+                patch_byte = Revision.get_from_revision_id(revision_id).patch_bytes
```

Comment:
Since `patch_bytes` is `nullable=True`, should we be checking to ensure this value isn't `None`? I'm not sure if it's possible for an empty patch to work here.

Technically `get_from_revision_id` can also be `None`, we should handle that case gracefully as well.

Reply:
> Since `patch_bytes` is `nullable=True`, should we be checking to ensure this value isn't `None`? I'm not sure if it's possible for an empty patch to work here.

So `patch_bytes` shouldn't be nullable, will change that (empty byte string should suffice as a default). The workflow is that there is no possibility of a landing job running without every revision having a patch (since those are updated in the original API call before submitting the job). I will add a couple of checks and comments to clarify that.

> Technically `get_from_revision_id` can also be `None`, we should handle that case gracefully as well.
For the same reasons as above, it would not be `None` once a landing job is created. I do have another (coming soon) patch that will change this relationship into a many to many, which will eliminate this issue here. I will add a comment.